src/slurm_cloud_bursting/resume.py

```diff
@@ -0,0 +1,262 @@
+        if config_file_path:
+            self._get_config(config_file_path)
+        else:
+            self.region = kwargs.get("region")
+            self.cluster_name = kwargs.get("cluster_name")
+            self.boto3_config = kwargs.get("boto3_config")
+            self.max_batch_size = kwargs.get("max_batch_size")
+            self.update_node_address = kwargs.get("update_node_address")
+
+    def __repr__(self):
+        attrs = ", ".join(["{key}={value}".format(key=key, value=repr(value)) for key, value in self.__dict__.items()])
+        return "{class_name}({attrs})".format(class_name=self.__class__.__name__, attrs=attrs)
+
+    def _get_config(self, config_file_path):
+        """Get resume program configuration."""
+        log.info("Reading %s", config_file_path)
+
+        config = ConfigParser()
+        try:
+            config.read_file(open(config_file_path, "r"))
+        except IOError:
+            log.error(f"Cannot read slurm cloud bursting scripts configuration file: {config_file_path}")
+            raise
+
+        self.region = config.get("slurm_cb_config", "region")
+        self.cluster_name = config.get("slurm_cb_config", "cluster_name")
+        self.max_batch_size = int(
+            config.get("slurm_cb_config", "max_batch_size", fallback=self.DEFAULT_MAX_INSTANCES_BATCH_SIZE)
+        )
+        self.update_node_address = config.getboolean("slurm_cb_config", "update_node_address", fallback=True)
+
+        # Configure boto3 to retry 5 times by default
+        self._boto3_config = {"retries": {"max_attempts": self.DEFAULT_MAX_RETRY, "mode": "standard"}}
+        proxy = config.get("slurm_cb_config", "proxy", fallback="NONE")
+        if proxy != "NONE":
+            self._boto3_config["proxies"] = {"https": proxy}
+        self.boto3_config = Config(**self._boto3_config)
+
+        log.info(self.__repr__())
+
+
+@retry(stop_max_attempt_number=3, wait_fixed=5000)
+def _set_nodes_down_and_power_save(node_list):
+    set_nodes_down(node_list, reason="Failure when resuming nodes")
+    set_nodes_power_down(node_list, reason="Failure when resuming nodes")
+
+
+def _handle_failed_nodes(node_list):
+    """
+    Fall back mechanism to handle failure when launching instances.
+
+    When encountering a failure, want slurm to deallocate current nodes,
+    and re-queue job to be run automatically by new nodes.
+    To do this, set node to DOWN, so slurm will automatically re-queue job.
+    Then set node to POWER_DOWN so suspend program will be run.
+    Suspend program needs to properly clean up instances(if any) and set node back to IDLE in all cases.
+
+    If this process is not done explicitly, slurm will wait until ResumeTimeout,
+    then execute this process of setting nodes to DOWN then POWER_DOWN.
+    To save time, should explicitly set nodes to DOWN then POWER_DOWN after encountering failure.
+    """
+    try:
+        log.info("Node %s marked as down and placed into power_down", node_list)
```

Comment:
nit: this is a list of nodes so probably we want to append it at the end of the log message